WcaOnRails/spec/requests/registrations_spec.rb

```diff
@@ -430,6 +430,212 @@
       end
     end
   end
+
+  describe "POST #process_payment_intent" do
+    context "when not signed in" do
+      let(:competition) { FactoryBot.create(:competition, :stripe_connected, :visible, :registration_open, events: Event.where(id: %w(222 333))) }
+      sign_out
+
+      it "redirects to the sign in page" do
+        # 1234 is just some random registration id
+        post registration_payment_intent_path(1234)
```

Comment:
What about creating a registration and using its id instead? ^^

Reply:
It doesn't matter since it shouldn't even look at the registration, but I can!
(The idea was to check it would do the authentication before anything else, and therefore not fail on invalid params but on an invalid authentication)